torchsynth/synth.py

```diff
@@ -79,13 +80,17 @@ def add_synth_modules(self, modules: List[Tuple[str, SynthModule]]):
 
             self.add_module(name, module)
 
-    def set_parameters(self, params: Dict[Tuple, T]):
+    def set_parameters(self, params: Dict[Tuple, T], freeze: Optional[bool] = False):
         """
-        Set parameters for synth by passing in a dictionary of modules and parameters
+        Set parameters for synth by passing in a dictionary of modules and parameters.
+        Can optionally freeze a parameter at this value to prevent further updates.
         """
         for (module_name, param_name), value in params.items():
             module = getattr(self, module_name)
             module.set_parameter(param_name, value.to(self.device))
+            # Freeze this parameter at this value now if freeze is True
+            if freeze:
+                module.get_parameter(param_name).frozen = True
```

Comment:
My proposal is that we add a convenience method

set_frozen_parameters where you just pass in a scalar and it turns it into the right batch size

That creates the tensor on self.device and calls self.set_parameters.

Reply:
Since in most cases you would probably want to freeze like that.

(Also, because then it give us a consistent naming scheme, like:

synth1B1-keyboard.midi_f0=69- 

even tho that's a little verbose

Reply:
I like this -- will add.

Reply:
Added just below